hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataAvroResolver.java

```diff
@@ -221,11 +215,41 @@ private static void validate(Schema schema, Map<QueryPath, MappingField> fieldsB
                 throw new IllegalArgumentException("Field '" + path + "' does not exist in schema");
             }
 
-            Schema.Type schemaFieldType = unwrapNullableType(schemaField.schema()).getType();
+            Schema fieldSchema = unwrapNullableType(schemaField.schema());
+            Schema.Type schemaFieldType = fieldSchema.getType();
+
+            if (mappingFieldTypeFamily == QueryDataTypeFamily.OBJECT) {
```

Comment:
nit: it would me more readable to move this up, after `if (conversions == null) {}` so `conversions` is determined in one place

Reply:
9a0e8973